### PR TITLE
feat(kno-7297): update MS Teams integration docs

### PR DESCRIPTION
**Pull request by mattmikolay**
### Description

This PR updates our Microsoft Teams integration docs. Just like what #643 did for our Slack docs, we’ll now have the following three pages for MS Teams:

- _Sending an internal message_
- _Sending a direct message_
- _Sending a message to channels_

The latter two pages describe sending a notification to Microsoft Teams as a bot, via Microsoft Bot Framework.

### Tasks

[KNO-7297](https://linear.app/knock/issue/KNO-7297/update-documentation-to-describe-sending-bot-messages-using-ms-teams)

### CODE REVIEW

File: content/integrations/chat/microsoft-teams/sending-a-direct-message.mdx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+---
+
+In this guide, we'll cover how to update a multi-tenant Microsoft Teams bot to send direct messages to Teams users using Knock. It assumes that you have already created a Microsoft Teams channel in Knock as outlined in the [Microsoft Teams integration overview](/integrations/chat/microsoft-teams/overview) guide.
+
+Here's what we'll cover in this guide:
+
+- Modeling a multi-tenant application in Knock using `Tenants`
+- Adding required scopes to your Microsoft Teams app's manifest
+- Setting required Tenant and User channel data when a Microsoft Teams bot is installed in a user's personal scope
+- Triggering a workflow with a user recipient to send a message to a DM
```
Comment (meryldakin) [inline content/integrations/chat/microsoft-teams/sending-a-direct-message.mdx:16]: I'm a little confused by the wording here, not sure if this is a typo or if I'm just not understanding it

File: content/integrations/chat/microsoft-teams/sending-a-direct-message.mdx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+## Prerequisites
+
+Make sure your bot has been registered with <a href="https://dev.botframework.com/" target="_blank">Microsoft Bot Framework</a> and is deployed to Azure. Knock does not manage deploying and configuring your bot. To set up Knock to send notifications as your bot, see [How to connect to Teams with Knock](/integrations/chat/microsoft-teams/overview#how-to-connect-to-teams-with-knock).
+
+## Key concepts
+
+`Tenants` are a concept you'll see throughout the following docs that are foundational to how Knock's Microsoft Teams integration works, but might not be used in every implementation of Knock.
```
Comment (meryldakin) [inline content/integrations/chat/microsoft-teams/sending-a-direct-message.mdx:24]: This part - `but might not be used in every implementation of Knock` - do we mean might not be used in every implementation of Knock wrt Teams or in general? It's a little unclear

File: content/integrations/chat/microsoft-teams/sending-a-direct-message.mdx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      "botId": "{{YOUR_BOT_ID}}",
+      "scopes": ["personal"]
+    }
+  ]
+}
+```
+
+## Setting Tenant and User channel data
+
+Your application should provide a method for admin users to set up your bot within a Microsoft Teams tenant. When this occurs, your backend should update the channel data associated with the Knock `Tenant`, passing in the `tenantId` of the Microsoft Teams tenant.
```
Comment (meryldakin) [inline content/integrations/chat/microsoft-teams/sending-a-direct-message.mdx:74]: do we mean `tenantId` or `ms_teams_tenant_id`? or if this is more of a generic "tenantId" maybe we don't put it in code format and just say 
```suggestion
Your application should provide a method for admin users to set up your bot within a Microsoft Teams tenant. When this occurs, your backend should update the channel data associated with the Knock `Tenant`, passing in the tenant ID of the Microsoft Teams tenant.
```

File: content/integrations/chat/microsoft-teams/sending-a-message-to-channels.mdx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+- Setting required Tenant and Object channel data when a Microsoft Teams bot is installed in a channel
+- Triggering a workflow with an object recipient to send a message to a Microsoft Teams channel
+
+## Prerequisites
+
+Make sure your bot has been registered with <a href="https://dev.botframework.com/" target="_blank">Microsoft Bot Framework</a> and is deployed to Azure. Knock does not manage deploying and configuring your bot. To set up Knock to send notifications as your bot, see [How to connect to Teams with Knock](/integrations/chat/microsoft-teams/overview#how-to-connect-to-teams-with-knock).
+
+## Key concepts
+
+There are two key concepts you'll see throughout the following docs that are foundational to how Knock's Microsoft Teams integration works, but might not be used in every implementation of Knock: tenants and objects.
```
Comment (meryldakin) [inline content/integrations/chat/microsoft-teams/sending-a-message-to-channels.mdx:24]:
```suggestion
There are two key concepts you'll see throughout the following docs that are foundational to how Knock's Microsoft Teams integration works: tenants and objects.
```
same suggestion as before, I don't think that caveat is giving any useful info and might be unnecessary to say

File: content/integrations/chat/microsoft-teams/sending-a-message-to-channels.mdx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+      that you use to power multi-tenancy in your application.
+    </>
+  }
+/>
+
+### About objects
+
+[Objects](/concepts/objects) in Knock are flexible abstractions meant to map to a resource in your system. Each individual object in Knock exists within a `collection` and requires an `id` unique to that collection.
+
+In the context of Knock's Microsoft Teams integration, objects serve two purposes. First, they store the Microsoft Teams channel or channels you want to notify. Second, they act as the recipient of the workflow that sends a message to Microsoft Teams.
```
Comment (meryldakin) [inline content/integrations/chat/microsoft-teams/sending-a-message-to-channels.mdx:49]:
```suggestion
In the context of Knock's Microsoft Teams integration, objects serve two purposes. First, they store the Microsoft Teams channel or channels you want to notify. Second, they act as the recipient of the workflow used to send a message to Microsoft Teams.
```

File: content/integrations/chat/microsoft-teams/sending-a-message-to-channels.mdx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+      You can think of collections as the different tables within your database that
+      represent resources in your application. Objects are the rows within those
+      tables.
+    </>
+  }
+/>
+
+#### Example
+
+Let's say we're building an example source control application like GitHub, where teams can collaborate and share code repositories. In this context, each GitHub organization would map to a tenant in Knock, and each repository would become an object inside of a `repositories` collection.
```
Comment (meryldakin) [inline content/integrations/chat/microsoft-teams/sending-a-message-to-channels.mdx:65]:
```suggestion
Let's say we're building a source control application like GitHub, where teams can collaborate and share code repositories. In this context, each GitHub organization would map to a tenant in Knock, and each repository would become an object inside of a `repositories` collection.
```

**Review by meryldakin — APPROVED** [review]
Nothing blocking! Just some light suggestions for clarity. Great job on these, they're very clear and succinct!

File: content/integrations/chat/microsoft-teams/sending-a-direct-message.mdx (hunk 1 above)
Comment (mattmikolay) [reply content/integrations/chat/microsoft-teams/sending-a-direct-message.mdx:16]: Good catch. I changed this to: “Triggering a workflow with a user recipient to send a direct message to a Teams user”.

**Pull request merged**